src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Dynamic.cs

```diff
@@ -1235,8 +1229,40 @@ public static void Main()
                 Signature(
                     "C+<>c", 
                     "<Main>b__0_0",
-                    ".method assembly hidebysig instance System.Object <Main>b__0_0([System.Runtime.CompilerServices.DynamicAttribute()] System.Object x) cil managed")
+                    ".method assembly hidebysig instance System.Object <Main>b__0_0([System.Runtime.CompilerServices.DynamicAttribute()] System.Object x, [System.Runtime.CompilerServices.DynamicAttribute(System.Collections.ObjectModel.ReadOnlyCollection`1[System.Reflection.CustomAttributeTypedArgument])] System.Object[] y) cil managed")
             });
         }
+
+        [Fact]
+        [WorkItem(6126, "https://github.com/dotnet/roslyn/issues/6126")]
+        public void DynamicLambdaParametersMissingBoolean()
+        {
+            var source0 =
+@"namespace System
+{
+    public class Object { }
+    public class ValueType { }
+    public struct Void { }
+    public struct IntPtr { }
+    public class MulticastDelegate { }
+}";
+            var source1 =
+@"delegate void D<T>(T t);
+class C
+{
+    static void Main()
+    {
+        D<dynamic[]> d = o => { };
+        d(null);
+    }
+}";
+            var comp = CreateCompilation(source0);
+            comp.VerifyDiagnostics();
+            var ref0 = comp.EmitToImageReference();
+            comp = CreateCompilation(source1, references: new[] { ref0, SystemCoreRef });
+            comp.VerifyDiagnostics();
+            // Make sure we emit without errors when System.Boolean is missing.
+            CompileAndVerify(comp, verify: false);
```

Comment:
I believe the intention was to not emit the `[DynamicAttribute]` in this case (and not generate any errors, allowing emit to succeed) since the attribute is only used in the debugger EE.